contrib/systemd-cloud-init.md

```diff
@@ -0,0 +1,322 @@
+
+You can tail the log to monitor the install progress:
+
+```
+tail -f /var/log/cloud-init-output.log 
+```
+
+Wait for the final `Good to go` message to know that the installer has finished.
+(Press Ctrl-C to exit from tail.)
+
+#### Check the status of the systemd service
+
+You can interact with systemd using `systemctl` (status, start, stop, restart,
+etc.) and `journalctl` (logging) tools.
+
+```
+systemctl status lbrycrd
+```
+
+```
+journalctl --unit lbrycrd
+```
+
+[Here is a tutorial to get you familiarized with
+systemd](https://www.digitalocean.com/community/tutorials/systemd-essentials-working-with-services-units-and-the-journal)
+
+#### Check the container
+
+You can get the same information directly from docker:
+
+```
+docker ps
+```
+
+```
+docker logs lbrycrd
+```
+
+### Utilize lbrycrd-cli
+
+You can use lbrycrd-cli from the host console. A bash alias has been added to
+`/root/.bash_aliases` that invokes lbrycrd-cli in its own container.
+
+```
+$ lbrycrd-cli getinfo
+{
+  "version": 120400,
+  "protocolversion": 70013,
+  "walletversion": 60000,
+  "balance": 0.00000000,
+  "blocks": 551965,
+  "timeoffset": 0,
+  "connections": 12,
+  "proxy": "",
+  "difficulty": 739465688254.7942,
+  "testnet": false,
+  "keypoololdest": 1555360604,
+  "keypoolsize": 101,
+  "paytxfee": 0.00000000,
+  "relayfee": 0.00001000,
+  "errors": ""
+}
+```
+
+### Chainquery (optional)
+
+The chainquery service is pre-installed, but it is not enabled by default.
+
+#### Enable and start the mysql service
+
+```
+systemctl enable --now mysql
+```
+
+In case you need it, there is a bash alias called `mysql`
+(`/root/.bash_aliases`) for the mysql client that allows you to login to the
+chainquery database.
+
+#### Enable and start the chainquery service
+
+The chainquery config file is located on the host: `/etc/lbry/chainqueryconfig.toml`
+
+```
+systemctl enable --now chainquery
+```
+
+In systemd, when you enable a service, it means to always start the service at
+system boot. (`--now` just means you also want to start the service right away.)
+
+As with any service, you can control chainquery with `systemctl` and get logs
+with `journalctl`:
+
+##### Starting and stopping chainquery service
+```
+systemctl start chainquery
+```
+
+```
+systemctl stop chainquery
+```
+
+##### Getting the chainquery service logs
+
+```
+journalctl --unit chainquery
+```
+
+(optionally use `-f` if you want to tail/follow the logs)
+
+##### Disabling chainquery service 
+
+```
+systemctl disable --now chainquery
+```
+
+### Known issues
+
+Ubuntu's snap update mechanism will apparently [restart docker even if there are
+no updates
+available](https://github.com/lbryio/lbry-docker/pull/50#issuecomment-485435736).
+In the future, this tutorial may replace the snap version of docker with the
+regular PPA version of docker-ce, which has a more predictable update strategy
+(apt-get) rather than auto-updates. More long term testing is needed to know
+which way is better.
+
```

Comment:
The fact that Snap does this is unbelievably irritating.  Good catch though this would be horrendous in live production and because of this issue I will recommending against people using Snap in production.